homeassistant/components/airzone/__init__.py

```diff
@@ -70,6 +82,47 @@ def get_airzone_value(self, key) -> Any:
         return value
 
 
+async def _async_migrate_unique_ids(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    coordinator: AirzoneUpdateCoordinator,
+) -> None:
+    """Migrate entities when the mac address gets discovered."""
+
+    @callback
+    def _async_migrator(entity_entry: er.RegistryEntry) -> dict[str, Any] | None:
+        updates = None
+
+        unique_id = entry.unique_id
+        entry_id = entry.entry_id
+        entity_unique_id = entity_entry.unique_id
+
+        if entity_unique_id.startswith(entry_id):
+            new_unique_id = f"{unique_id}{entity_unique_id[len(entry_id):]}"
+            _LOGGER.info(
+                "Migrating unique_id from [%s] to [%s]",
+                entity_unique_id,
+                new_unique_id,
+            )
+            updates = {"new_unique_id": new_unique_id}
+
+        return updates
+
+    if (
+        entry.unique_id is None
+        and AZD_WEBSERVER in coordinator.data
+        and AZD_MAC in coordinator.data[AZD_WEBSERVER]
+    ):
+        mac = coordinator.data[AZD_WEBSERVER][AZD_MAC]
+        if mac is not None:
```

Comment:
```suggestion
        if (mac := coordinator.data[AZD_WEBSERVER][AZD_MAC]) is None:
            return
```

You could even move it into the conditional above, reverse it and return early instead

Reply:
You suggestion reduced the code coverage, so I merged everything in the if block b60eb601666f361c03e4386677af4f4395e4cf41
I hope that's OK.